example/app/urls.py

```diff
@@ -1,16 +1,13 @@
 from django.contrib.auth import views as auth_views
-try:
-    from django.urls import include, url
-except ImportError:
-    from django.conf.urls import include, url
+from django.urls import include, re_path
 from django.contrib import admin
 from django.views.generic import TemplateView
 
 
 urlpatterns = [
-    url(r'^$', TemplateView.as_view(template_name='home.html'), name='home'),
-    url(r'^accounts/login/$', auth_views.login, {'template_name': 'login.html'}, name='login'),
-    url(r'^accounts/logout/$', auth_views.logout, {'next_page': '/'}, name='logout'),
-    url(r'^', include('oidc_provider.urls', namespace='oidc_provider')),
-    url(r'^admin/', admin.site.urls),
+    re_path(r'^$', TemplateView.as_view(template_name='home.html'), name='home'),
+    re_path(r'^accounts/login/$', auth_views.login, {'template_name': 'login.html'}, name='login'),
+    re_path(r'^accounts/logout/$', auth_views.logout, {'next_page': '/'}, name='logout'),
+    re_path(r'^', include(('oidc_provider.urls', 'oidc_provider'), namespace='oidc_provider')),
+    re_path(r'^admin/', admin.site.urls),
```

Comment:
**Suggestion:** In Django 3.1, the login and logout views have been updated to LoginView and LogoutView respectively. Update the auth_views.login and auth_views.logout to auth_views.LoginView.as_view() and auth_views.LogoutView.as_view() respectively.
```suggestion
    urlpatterns = [
        re_path(r'^$', TemplateView.as_view(template_name='home.html'), name='home'),
        re_path(r'^accounts/login/$', auth_views.LoginView.as_view(template_name='login.html'), name='login'),
        re_path(r'^accounts/logout/$', auth_views.LogoutView.as_view(next_page='/'), name='logout'),
        re_path(r'^', include(('oidc_provider.urls', 'oidc_provider'), namespace='oidc_provider')),
        re_path(r'^admin/', admin.site.urls),
    ]
```